### PR TITLE
feat(document-model-libs): address initial QA feedback

### PR DESCRIPTION
**Pull request by ryanwolhuter**
Address initial QA feedback:

- make inputs submit individually on blur and on press enter
- add try catch to prevent error on broken schema
- use light mode editor theme
- add initial layout styles
- use auto-expanding textarea for input fields
- show schema errors underneath text editor
- remove `setName` call from model name field
- fix wrong value used for document type

### CODE REVIEW

File: packages/document-model-libs/editors/document-model-2/components/json-editor.tsx [hunk 1]
```diff
@@ -56,8 +57,7 @@ export function JSONEditor(props: Props) {
         EditorView.focusChangeEffect.of((state, focusing) => {
           if (readonly || focusing) return null;
           const newDoc = state.doc.toString();
-          const isValid = jsonLinter(viewRef.current!).length === 0;
-          if (!!newDoc && isValid && newDoc !== doc) {
+          if (!!newDoc && newDoc !== doc) {
```
Comment (ryanwolhuter) [inline packages/document-model-libs/editors/document-model-2/components/json-editor.tsx:60]: focus change always dispatches even if there are errors

File: packages/document-model-libs/editors/document-model-2/components/text-area.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import * as React from "react";
+import { cn } from "../utils";
+
+export interface TextareaProps
+  extends React.TextareaHTMLAttributes<HTMLTextAreaElement> {}
+
+const Textarea = React.forwardRef<HTMLTextAreaElement, TextareaProps>(
```
Comment (ryanwolhuter) [inline packages/document-model-libs/editors/document-model-2/components/text-area.tsx:7]: this textarea component grows with the text as requested

File: packages/document-model-libs/editors/document-model-2/document-model-editor.tsx [hunk 3]
```diff
@@ -16,6 +16,7 @@ type Props = {
   localStateInitialValue: string;
   handlers: DocumentActionHandlers;
   modules: Module[];
+  errors: string;
```
Comment (ryanwolhuter) [inline packages/document-model-libs/editors/document-model-2/document-model-editor.tsx:19]: functionality of the component is the same, but layout has been adjusted. also shows the errors under the editor

File: packages/document-model-libs/editors/document-model-2/editor.tsx [hunk 4]
```diff
@@ -59,18 +52,30 @@ export default function Editor(
...
-      )
-      .filter(Boolean)
-      .join("\n")}
-  `);
+  const [schema, setSchema] = useState(initialSchema);
+  const [errors, setErrors] = useState("");
+
+  useEffect(() => {
+    try {
+      const newSchema = buildSchema(`
```
Comment (ryanwolhuter) [inline packages/document-model-libs/editors/document-model-2/editor.tsx:60]: `buildSchema` throws on invalid input. use try catch and show errors in component

File: packages/document-model-libs/editors/document-model-2/utils/helpers.ts [hunk 5]
```diff
@@ -111,8 +111,8 @@ export function getDocumentMetadata(document: DocumentModelDocument) {
     website: globalState.author.website,
   };
   return {
-    name: document.name,
-    documentType: document.documentType,
+    name: globalState.name,
```
Comment (ryanwolhuter) [inline packages/document-model-libs/editors/document-model-2/utils/helpers.ts:114]: use global state name and id for these values

**Review by acaldas — APPROVED** [review]

**Pull request merged**